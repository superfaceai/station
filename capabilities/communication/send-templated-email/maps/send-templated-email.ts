/* eslint-disable jest/no-export */

import { SuperfaceTest } from '@superfaceai/testing-lib';

export const sendTemplatedEmailTest = (
  provider: string,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  params: { from: string; to: string; templateId: string; [key: string]: any }
): void => {
  describe(`communication/send-templated-email/${provider}`, () => {
    let superface: SuperfaceTest;

    beforeEach(async () => {
      superface = new SuperfaceTest();
    });

    describe('SendTemplatedEmail', () => {
      describe('when all inputs are correct', () => {
        it('should return messageId as result', async () => {
          const input = {
            ...params,
            templateData: params,
          };

          await expect(
            superface.run({
              profile: 'communication/send-templated-email',
              provider,
              useCase: 'SendTemplatedEmail',
              input,
            })
          ).resolves.toMatchSnapshot();
        });
      });

      describe('when inputs are invalid', () => {
        it('should throw error', async () => {
          const input = {
            ...params,
            to: 'invalidemail',
            from: 'invalidemail',
            templateId: 'invalid',
            templateData: undefined,
          };

          await expect(
            superface.run({
              profile: 'communication/send-templated-email',
              provider,
              useCase: 'SendTemplatedEmail',
              input,
            })
          ).resolves.toMatchSnapshot();
        });
      });
    });
  });
};
