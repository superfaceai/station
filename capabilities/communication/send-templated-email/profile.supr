"""
Send Templated Email
Send templated transactional email to one recipient.
"""

name = "communication/send-templated-email"
version = "1.0.1"

"""
Send Templated Email
Send templated transactional email to one recipient.
Requires template defined on provider side. 
"""
usecase SendTemplatedEmail unsafe {
  input {
    """
    From
    The sender's email address.
    """
    from!

    """
    To
    The recipient's email address.
    """
    to!

    """
    Template Identifier
    The template to use when sending email message.
    """
    templateId!

    """
    Template Data
    Template data to be applied to the specified template to generate html, test, and subject.
    The value is a collection of key/value pairs following the pattern `variable_name`: `value to insert`.
    This field should be used in combination with `templateId` to identify what template to use.
    """
    templateData!
  }

  result {
    """
    Message Identifier
    The identifier is provider-specific and not unique.
    """
    messageId!
  }

  error {
    """
    Title
    A short, human-readable summary of the problem type.
    """
    title!

    """
    A human-readable explanation specific to this occurrence of the problem.
    """
    detail
  }

  example Successful {
    input {
      from = 'no-reply@example.com',
      to = 'jane.doe@example.com',
      templateId = 'user-invitation',
      templateData = {
        invited_by = 'Bob',
        invitation_link = 'example.com/invite?code=invite',
      },
    }

    result {
      messageId = 'gyax6ce9QLKQ80VOqsRX2g',
    }
  }

  example Failed {
    input {
      from = 'no-reply@example.com',
      to = 'Jane Doe',
      templateId = 'template-id',
      templateData = {
        template_data_param = 'value',
      },
    }

    error {
      title = 'Invalid inputs',
      detail = 'An email address must contain a single @',
    }
  }
}
