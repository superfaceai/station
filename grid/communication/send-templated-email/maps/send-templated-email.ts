/* eslint-disable jest/no-export */

import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';

import { buildSuperfaceTest } from '../../../test-config';

export const sendTemplatedEmailTest = (
  provider: string,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  params: { from: string; to: string; templateId: string; [key: string]: any },
  options?: RecordingProcessOptions
): void => {
  describe(`communication/send-templated-email/${provider}`, () => {
    let superface: SuperfaceTest;

    beforeEach(async () => {
      superface = buildSuperfaceTest({
        profile: 'communication/send-templated-email',
        provider,
      });
    });

    describe('SendTemplatedEmail', () => {
      describe('when all inputs are correct', () => {
        it('should return messageId as result', async () => {
          const input = {
            ...params,
            templateData: params,
          };

          await expect(
            superface.run(
              {
                useCase: 'SendTemplatedEmail',
                input,
              },
              options
            )
          ).resolves.toMatchSnapshot();
        });
      });

      describe('when inputs are invalid', () => {
        it('should throw error', async () => {
          const input = {
            ...params,
            to: 'invalidemail',
            from: 'invalidemail',
            templateId: 'invalid',
            templateData: undefined,
          };

          await expect(
            superface.run(
              {
                useCase: 'SendTemplatedEmail',
                input,
              },
              options
            )
          ).resolves.toMatchSnapshot();
        });
      });
    });
  });
};
