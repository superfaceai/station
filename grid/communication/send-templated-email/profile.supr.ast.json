{
  "kind": "ProfileDocument",
  "header": {
    "kind": "ProfileHeader",
    "scope": "communication",
    "name": "send-templated-email",
    "version": {
      "major": 1,
      "minor": 0,
      "patch": 1
    },
    "location": {
      "start": {
        "line": 6,
        "column": 1,
        "charIndex": 83
      },
      "end": {
        "line": 7,
        "column": 18,
        "charIndex": 144
      }
    },
    "documentation": {
      "title": "Send Templated Email",
      "description": "Send templated transactional email to one recipient.",
      "location": {
        "start": {
          "line": 1,
          "column": 1,
          "charIndex": 0
        },
        "end": {
          "line": 4,
          "column": 4,
          "charIndex": 81
        }
      }
    }
  },
  "definitions": [
    {
      "kind": "UseCaseDefinition",
      "useCaseName": "SendTemplatedEmail",
      "safety": "unsafe",
      "input": {
        "kind": "UseCaseSlotDefinition",
        "value": {
          "kind": "ObjectDefinition",
          "fields": [
            {
              "kind": "FieldDefinition",
              "fieldName": "from",
              "required": true,
              "location": {
                "start": {
                  "line": 20,
                  "column": 5,
                  "charIndex": 380
                },
                "end": {
                  "line": 20,
                  "column": 10,
                  "charIndex": 385
                }
              },
              "documentation": {
                "title": "From",
                "description": "The sender's email address.",
                "location": {
                  "start": {
                    "line": 16,
                    "column": 5,
                    "charIndex": 323
                  },
                  "end": {
                    "line": 19,
                    "column": 8,
                    "charIndex": 375
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "to",
              "required": true,
              "location": {
                "start": {
                  "line": 26,
                  "column": 5,
                  "charIndex": 449
                },
                "end": {
                  "line": 26,
                  "column": 8,
                  "charIndex": 452
                }
              },
              "documentation": {
                "title": "To",
                "description": "The recipient's email address.",
                "location": {
                  "start": {
                    "line": 22,
                    "column": 5,
                    "charIndex": 391
                  },
                  "end": {
                    "line": 25,
                    "column": 8,
                    "charIndex": 444
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "templateId",
              "required": true,
              "location": {
                "start": {
                  "line": 32,
                  "column": 5,
                  "charIndex": 550
                },
                "end": {
                  "line": 32,
                  "column": 16,
                  "charIndex": 561
                }
              },
              "documentation": {
                "title": "Template Identifier",
                "description": "The template to use when sending email message.",
                "location": {
                  "start": {
                    "line": 28,
                    "column": 5,
                    "charIndex": 458
                  },
                  "end": {
                    "line": 31,
                    "column": 8,
                    "charIndex": 545
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "templateData",
              "required": true,
              "location": {
                "start": {
                  "line": 40,
                  "column": 5,
                  "charIndex": 900
                },
                "end": {
                  "line": 40,
                  "column": 18,
                  "charIndex": 913
                }
              },
              "documentation": {
                "title": "Template Data",
                "description": "Template data to be applied to the specified template to generate html, test, and subject.\nThe value is a collection of key/value pairs following the pattern `variable_name`: `value to insert`.\nThis field should be used in combination with `templateId` to identify what template to use.",
                "location": {
                  "start": {
                    "line": 34,
                    "column": 5,
                    "charIndex": 567
                  },
                  "end": {
                    "line": 39,
                    "column": 8,
                    "charIndex": 895
                  }
                }
              }
            }
          ],
          "location": {
            "start": {
              "line": 15,
              "column": 9,
              "charIndex": 317
            },
            "end": {
              "line": 41,
              "column": 4,
              "charIndex": 917
            }
          }
        },
        "location": {
          "start": {
            "line": 15,
            "column": 3,
            "charIndex": 311
          },
          "end": {
            "line": 41,
            "column": 4,
            "charIndex": 917
          }
        }
      },
      "result": {
        "kind": "UseCaseSlotDefinition",
        "value": {
          "kind": "ObjectDefinition",
          "fields": [
            {
              "kind": "FieldDefinition",
              "fieldName": "messageId",
              "required": true,
              "location": {
                "start": {
                  "line": 48,
                  "column": 5,
                  "charIndex": 1029
                },
                "end": {
                  "line": 48,
                  "column": 15,
                  "charIndex": 1039
                }
              },
              "documentation": {
                "title": "Message Identifier",
                "description": "The identifier is provider-specific and not unique.",
                "location": {
                  "start": {
                    "line": 44,
                    "column": 5,
                    "charIndex": 934
                  },
                  "end": {
                    "line": 47,
                    "column": 8,
                    "charIndex": 1024
                  }
                }
              }
            }
          ],
          "location": {
            "start": {
              "line": 43,
              "column": 10,
              "charIndex": 928
            },
            "end": {
              "line": 49,
              "column": 4,
              "charIndex": 1043
            }
          }
        },
        "location": {
          "start": {
            "line": 43,
            "column": 3,
            "charIndex": 921
          },
          "end": {
            "line": 49,
            "column": 4,
            "charIndex": 1043
          }
        }
      },
      "error": {
        "kind": "UseCaseSlotDefinition",
        "value": {
          "kind": "ObjectDefinition",
          "fields": [
            {
              "kind": "FieldDefinition",
              "fieldName": "title",
              "required": true,
              "location": {
                "start": {
                  "line": 56,
                  "column": 5,
                  "charIndex": 1142
                },
                "end": {
                  "line": 56,
                  "column": 11,
                  "charIndex": 1148
                }
              },
              "documentation": {
                "title": "Title",
                "description": "A short, human-readable summary of the problem type.",
                "location": {
                  "start": {
                    "line": 52,
                    "column": 5,
                    "charIndex": 1059
                  },
                  "end": {
                    "line": 55,
                    "column": 8,
                    "charIndex": 1137
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "detail",
              "required": false,
              "location": {
                "start": {
                  "line": 61,
                  "column": 5,
                  "charIndex": 1247
                },
                "end": {
                  "line": 61,
                  "column": 11,
                  "charIndex": 1253
                }
              },
              "documentation": {
                "title": "A human-readable explanation specific to this occurrence of the problem.",
                "location": {
                  "start": {
                    "line": 58,
                    "column": 5,
                    "charIndex": 1154
                  },
                  "end": {
                    "line": 60,
                    "column": 8,
                    "charIndex": 1242
                  }
                }
              }
            }
          ],
          "location": {
            "start": {
              "line": 51,
              "column": 9,
              "charIndex": 1053
            },
            "end": {
              "line": 62,
              "column": 4,
              "charIndex": 1257
            }
          }
        },
        "location": {
          "start": {
            "line": 51,
            "column": 3,
            "charIndex": 1047
          },
          "end": {
            "line": 62,
            "column": 4,
            "charIndex": 1257
          }
        }
      },
      "examples": [
        {
          "kind": "UseCaseSlotDefinition",
          "value": {
            "kind": "UseCaseExample",
            "exampleName": "Successful",
            "input": {
              "kind": "UseCaseSlotDefinition",
              "value": {
                "kind": "ComlinkObjectLiteral",
                "fields": [
                  {
                    "kind": "ComlinkAssignment",
                    "key": ["from"],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "no-reply@example.com",
                      "location": {
                        "start": {
                          "line": 66,
                          "column": 14,
                          "charIndex": 1307
                        },
                        "end": {
                          "line": 66,
                          "column": 36,
                          "charIndex": 1329
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 66,
                        "column": 7,
                        "charIndex": 1300
                      },
                      "end": {
                        "line": 66,
                        "column": 36,
                        "charIndex": 1329
                      }
                    }
                  },
                  {
                    "kind": "ComlinkAssignment",
                    "key": ["to"],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "jane.doe@example.com",
                      "location": {
                        "start": {
                          "line": 67,
                          "column": 12,
                          "charIndex": 1342
                        },
                        "end": {
                          "line": 67,
                          "column": 34,
                          "charIndex": 1364
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 67,
                        "column": 7,
                        "charIndex": 1337
                      },
                      "end": {
                        "line": 67,
                        "column": 34,
                        "charIndex": 1364
                      }
                    }
                  },
                  {
                    "kind": "ComlinkAssignment",
                    "key": ["templateId"],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "user-invitation",
                      "location": {
                        "start": {
                          "line": 68,
                          "column": 20,
                          "charIndex": 1385
                        },
                        "end": {
                          "line": 68,
                          "column": 37,
                          "charIndex": 1402
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 68,
                        "column": 7,
                        "charIndex": 1372
                      },
                      "end": {
                        "line": 68,
                        "column": 37,
                        "charIndex": 1402
                      }
                    }
                  },
                  {
                    "kind": "ComlinkAssignment",
                    "key": ["templateData"],
                    "value": {
                      "kind": "ComlinkObjectLiteral",
                      "fields": [
                        {
                          "kind": "ComlinkAssignment",
                          "key": ["invited_by"],
                          "value": {
                            "kind": "ComlinkPrimitiveLiteral",
                            "value": "Bob",
                            "location": {
                              "start": {
                                "line": 70,
                                "column": 22,
                                "charIndex": 1448
                              },
                              "end": {
                                "line": 70,
                                "column": 27,
                                "charIndex": 1453
                              }
                            }
                          },
                          "location": {
                            "start": {
                              "line": 70,
                              "column": 9,
                              "charIndex": 1435
                            },
                            "end": {
                              "line": 70,
                              "column": 27,
                              "charIndex": 1453
                            }
                          }
                        },
                        {
                          "kind": "ComlinkAssignment",
                          "key": ["invitation_link"],
                          "value": {
                            "kind": "ComlinkPrimitiveLiteral",
                            "value": "example.com/invite?code=invite",
                            "location": {
                              "start": {
                                "line": 71,
                                "column": 27,
                                "charIndex": 1481
                              },
                              "end": {
                                "line": 71,
                                "column": 59,
                                "charIndex": 1513
                              }
                            }
                          },
                          "location": {
                            "start": {
                              "line": 71,
                              "column": 9,
                              "charIndex": 1463
                            },
                            "end": {
                              "line": 71,
                              "column": 59,
                              "charIndex": 1513
                            }
                          }
                        }
                      ],
                      "location": {
                        "start": {
                          "line": 69,
                          "column": 22,
                          "charIndex": 1425
                        },
                        "end": {
                          "line": 72,
                          "column": 8,
                          "charIndex": 1522
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 69,
                        "column": 7,
                        "charIndex": 1410
                      },
                      "end": {
                        "line": 72,
                        "column": 8,
                        "charIndex": 1522
                      }
                    }
                  }
                ],
                "location": {
                  "start": {
                    "line": 65,
                    "column": 11,
                    "charIndex": 1292
                  },
                  "end": {
                    "line": 73,
                    "column": 6,
                    "charIndex": 1529
                  }
                }
              },
              "location": {
                "start": {
                  "line": 65,
                  "column": 5,
                  "charIndex": 1286
                },
                "end": {
                  "line": 73,
                  "column": 6,
                  "charIndex": 1529
                }
              }
            },
            "result": {
              "kind": "UseCaseSlotDefinition",
              "value": {
                "kind": "ComlinkObjectLiteral",
                "fields": [
                  {
                    "kind": "ComlinkAssignment",
                    "key": ["messageId"],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "gyax6ce9QLKQ80VOqsRX2g",
                      "location": {
                        "start": {
                          "line": 76,
                          "column": 19,
                          "charIndex": 1562
                        },
                        "end": {
                          "line": 76,
                          "column": 43,
                          "charIndex": 1586
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 76,
                        "column": 7,
                        "charIndex": 1550
                      },
                      "end": {
                        "line": 76,
                        "column": 43,
                        "charIndex": 1586
                      }
                    }
                  }
                ],
                "location": {
                  "start": {
                    "line": 75,
                    "column": 12,
                    "charIndex": 1542
                  },
                  "end": {
                    "line": 77,
                    "column": 6,
                    "charIndex": 1593
                  }
                }
              },
              "location": {
                "start": {
                  "line": 75,
                  "column": 5,
                  "charIndex": 1535
                },
                "end": {
                  "line": 77,
                  "column": 6,
                  "charIndex": 1593
                }
              }
            },
            "location": {
              "start": {
                "line": 64,
                "column": 11,
                "charIndex": 1269
              },
              "end": {
                "line": 78,
                "column": 4,
                "charIndex": 1597
              }
            }
          },
          "location": {
            "start": {
              "line": 64,
              "column": 3,
              "charIndex": 1261
            },
            "end": {
              "line": 78,
              "column": 4,
              "charIndex": 1597
            }
          }
        },
        {
          "kind": "UseCaseSlotDefinition",
          "value": {
            "kind": "UseCaseExample",
            "exampleName": "Failed",
            "input": {
              "kind": "UseCaseSlotDefinition",
              "value": {
                "kind": "ComlinkObjectLiteral",
                "fields": [
                  {
                    "kind": "ComlinkAssignment",
                    "key": ["from"],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "no-reply@example.com",
                      "location": {
                        "start": {
                          "line": 82,
                          "column": 14,
                          "charIndex": 1643
                        },
                        "end": {
                          "line": 82,
                          "column": 36,
                          "charIndex": 1665
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 82,
                        "column": 7,
                        "charIndex": 1636
                      },
                      "end": {
                        "line": 82,
                        "column": 36,
                        "charIndex": 1665
                      }
                    }
                  },
                  {
                    "kind": "ComlinkAssignment",
                    "key": ["to"],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "Jane Doe",
                      "location": {
                        "start": {
                          "line": 83,
                          "column": 12,
                          "charIndex": 1678
                        },
                        "end": {
                          "line": 83,
                          "column": 22,
                          "charIndex": 1688
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 83,
                        "column": 7,
                        "charIndex": 1673
                      },
                      "end": {
                        "line": 83,
                        "column": 22,
                        "charIndex": 1688
                      }
                    }
                  },
                  {
                    "kind": "ComlinkAssignment",
                    "key": ["templateId"],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "template-id",
                      "location": {
                        "start": {
                          "line": 84,
                          "column": 20,
                          "charIndex": 1709
                        },
                        "end": {
                          "line": 84,
                          "column": 33,
                          "charIndex": 1722
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 84,
                        "column": 7,
                        "charIndex": 1696
                      },
                      "end": {
                        "line": 84,
                        "column": 33,
                        "charIndex": 1722
                      }
                    }
                  },
                  {
                    "kind": "ComlinkAssignment",
                    "key": ["templateData"],
                    "value": {
                      "kind": "ComlinkObjectLiteral",
                      "fields": [
                        {
                          "kind": "ComlinkAssignment",
                          "key": ["template_data_param"],
                          "value": {
                            "kind": "ComlinkPrimitiveLiteral",
                            "value": "value",
                            "location": {
                              "start": {
                                "line": 86,
                                "column": 31,
                                "charIndex": 1777
                              },
                              "end": {
                                "line": 86,
                                "column": 38,
                                "charIndex": 1784
                              }
                            }
                          },
                          "location": {
                            "start": {
                              "line": 86,
                              "column": 9,
                              "charIndex": 1755
                            },
                            "end": {
                              "line": 86,
                              "column": 38,
                              "charIndex": 1784
                            }
                          }
                        }
                      ],
                      "location": {
                        "start": {
                          "line": 85,
                          "column": 22,
                          "charIndex": 1745
                        },
                        "end": {
                          "line": 87,
                          "column": 8,
                          "charIndex": 1793
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 85,
                        "column": 7,
                        "charIndex": 1730
                      },
                      "end": {
                        "line": 87,
                        "column": 8,
                        "charIndex": 1793
                      }
                    }
                  }
                ],
                "location": {
                  "start": {
                    "line": 81,
                    "column": 11,
                    "charIndex": 1628
                  },
                  "end": {
                    "line": 88,
                    "column": 6,
                    "charIndex": 1800
                  }
                }
              },
              "location": {
                "start": {
                  "line": 81,
                  "column": 5,
                  "charIndex": 1622
                },
                "end": {
                  "line": 88,
                  "column": 6,
                  "charIndex": 1800
                }
              }
            },
            "error": {
              "kind": "UseCaseSlotDefinition",
              "value": {
                "kind": "ComlinkObjectLiteral",
                "fields": [
                  {
                    "kind": "ComlinkAssignment",
                    "key": ["title"],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "Invalid inputs",
                      "location": {
                        "start": {
                          "line": 91,
                          "column": 15,
                          "charIndex": 1828
                        },
                        "end": {
                          "line": 91,
                          "column": 31,
                          "charIndex": 1844
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 91,
                        "column": 7,
                        "charIndex": 1820
                      },
                      "end": {
                        "line": 91,
                        "column": 31,
                        "charIndex": 1844
                      }
                    }
                  },
                  {
                    "kind": "ComlinkAssignment",
                    "key": ["detail"],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "An email address must contain a single @",
                      "location": {
                        "start": {
                          "line": 92,
                          "column": 16,
                          "charIndex": 1861
                        },
                        "end": {
                          "line": 92,
                          "column": 58,
                          "charIndex": 1903
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 92,
                        "column": 7,
                        "charIndex": 1852
                      },
                      "end": {
                        "line": 92,
                        "column": 58,
                        "charIndex": 1903
                      }
                    }
                  }
                ],
                "location": {
                  "start": {
                    "line": 90,
                    "column": 11,
                    "charIndex": 1812
                  },
                  "end": {
                    "line": 93,
                    "column": 6,
                    "charIndex": 1910
                  }
                }
              },
              "location": {
                "start": {
                  "line": 90,
                  "column": 5,
                  "charIndex": 1806
                },
                "end": {
                  "line": 93,
                  "column": 6,
                  "charIndex": 1910
                }
              }
            },
            "location": {
              "start": {
                "line": 80,
                "column": 11,
                "charIndex": 1609
              },
              "end": {
                "line": 94,
                "column": 4,
                "charIndex": 1914
              }
            }
          },
          "location": {
            "start": {
              "line": 80,
              "column": 3,
              "charIndex": 1601
            },
            "end": {
              "line": 94,
              "column": 4,
              "charIndex": 1914
            }
          }
        }
      ],
      "location": {
        "start": {
          "line": 14,
          "column": 1,
          "charIndex": 273
        },
        "end": {
          "line": 95,
          "column": 2,
          "charIndex": 1916
        }
      },
      "documentation": {
        "title": "Send Templated Email",
        "description": "Send templated transactional email to one recipient.\nRequires template defined on provider side.",
        "location": {
          "start": {
            "line": 9,
            "column": 1,
            "charIndex": 146
          },
          "end": {
            "line": 13,
            "column": 4,
            "charIndex": 272
          }
        }
      }
    }
  ],
  "location": {
    "start": {
      "line": 6,
      "column": 1,
      "charIndex": 83
    },
    "end": {
      "line": 95,
      "column": 2,
      "charIndex": 1916
    }
  },
  "astMetadata": {
    "astVersion": {
      "major": 1,
      "minor": 2,
      "patch": 0
    },
    "parserVersion": {
      "major": 1,
      "minor": 2,
      "patch": 0
    },
    "sourceChecksum": "f919f7577d8acd2dd2acbe8b9e03d3f7083eae8c38c0f889418e19ac2cd049c4"
  }
}
