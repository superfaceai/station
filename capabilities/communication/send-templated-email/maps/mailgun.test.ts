import { sendTemplatedEmailTest } from './send-templated-email';

sendTemplatedEmailTest('mailgun', {
  from: 'demo@demo.superface.org',
  to: 'demo@superface.ai',
  templateId: 'station-test',
});
