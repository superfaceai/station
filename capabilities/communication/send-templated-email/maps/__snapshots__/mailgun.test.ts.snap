// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`communication/send-templated-email/mailgun SendTemplatedEmail when all inputs are correct should return messageId as result 1`] = `
Object {
  "value": Object {
    "messageId": "<20211102124741.1.1F208F68EE6FA148@demo.superface.org>",
  },
}
`;

exports[`communication/send-templated-email/mailgun SendTemplatedEmail when inputs are invalid should throw error 1`] = `
Object {
  "error": "HTTPError: Expected HTTP error
Properites: {
  \\"title\\": \\"Invalid inputs\\",
  \\"message\\": \\"'from' parameter is not a valid address. please check documentation\\"
}
AST Path: definitions[0].statements[0].responseHandlers[1].statements[0]
Original Map Location: Line 27, column 7",
}
`;
